### ISSUE TITLE
Some Problems

### ISSUE THREAD

**Issue by 747929791** [matthew-z/R-net#2]
Hi,
I found some problems when I try to run this code.
When I use "python main.py" to run this demo, I got these in the final result:
  step 13460 / 1347 of epoch 9)
  loss:  2.74846293926239
  acc:  0.509375
  speed: 7.808003 examples/sec 

  exact_match: 0.983917)
  f1: 6.878085)

When I use "python main.py --batch_size 32" to run this demo, I got these in the final result:
  step 26920 / 2693 of epoch 9)
  loss:  2.330745005607605
  acc:  0.55
  speed: 4.082112 examples/sec 

  exact_match: 0.794702)
  f1: 7.338743)

I don't understand why the exact_match and f1 are so small.(exact_match=0.79% f1=7.3%?) I think that this is much smaller than the results of the paper. Did I do something wrong?

**Comment by matthew-z** [comment]
Sorry, there are some problems in the model. I am still trying to figure out what the problems are.

Alternatively, there is one simplified version in pytorch (no Bi-RNN, no multi-layer):[https://github.com/akaitsuki-ii/r-net](https://github.com/akaitsuki-ii/r-net), but it has the same problem (very low performance on validation data).

One in Tensorflow: [https://github.com/minsangkim142/R-net](https://github.com/minsangkim142/R-net). This one is much better than mine: it got about 50 on F1.

**Comment by matthew-z** [comment]
FYI: I have figured out why the performance is so slow. I will push a fix in a few days.

**Comment by qixianbiao** [comment]
Can you point out the reason why the evaluated performance is low? You mentioned you fixed it. Thanks in advance.

**Comment by matthew-z** [comment]
@qixianbiao 

I've figured out that there were some bugs in numeralizing the dataset and attention. 
Fixing them can make the performance much better, but the result is still lower than paper's (I haven't figured out why)

**Comment by qixianbiao** [comment]
@matthew-z  
Thanks for your response. Do you mean it is because of the preprocessing issue?
There are tensorflow and keras verisons of R-Net, their performance is around EM/F1 of 55/67, but I am fan of pytorch. 
Have read your code, good style. Thanks for your contribution.

Can you merge your new update into the repository? Thanks.
I am still new to NLP.

**Comment by matthew-z** [comment]
@qixianbiao Sure, I plan to do it by this weekend.

**Comment by qixianbiao** [comment]
Thanks a lot.

**Comment by matthew-z** [comment]
@qixianbiao I have pushed an update. Currently its performance is around EM 55 / F1 65 without char-level embedding after 4 hours training on GTX 1070
